server.js

```diff
@@ -171,23 +180,33 @@ function addrport(s) {
         return { host: spl[0], port: parseInt(spl[1]) };
 }
 
-if (config.http.enabled) {
-    var ap = addrport(config.http.listen);
-    var server = http.createServer(app).listen(ap.port, ap.host, function() {
-        var host = server.address().address;
-        var port = server.address().port;
-        console.log('Started server at http://%s:%s', host, port);
-    });
+function serv(server, serverconfig, callback) {
+  var ap = addrport(serverconfig.listen);
+  return server.listen(ap.port, ap.host, callback);
 }
-if (config.https.enabled) {
-    var sec_ap = addrport(config.https.listen);
-    var sec_creds = {
-        key: fs.readFileSync(config.https.key),
-        cert: fs.readFileSync(config.https.cert)
-    };
-    var sec_server = https.createServer(sec_creds, app).listen(sec_ap.port, sec_ap.host, function() {
-        var host = server.address().address;
-        var port = server.address().port;
-        console.log('Started server at https://%s:%s', host, port);
+
+function init(config) {
```

Comment:
Why do we have init and main? This should just be one single function (preferably main), and the readFileSync should be merged into it.